### ISSUE TITLE
should move sensitive configs into local.js to prevent them from being in the repos

### ISSUE THREAD

**Issue by mikedevita** [ghaiklor/generator-sails-rest-api#301]
I will be submitting another PR to resolve this if you like, basically in the api/services/* files you do something like this.. this can be applied for things like db name, user, host and password.

```javascript
const _      = require('lodash')
const config = _.merge(require('../../config/services/mailer'), require('../../config/local'));
```

and create a config/local.js with something like this....

```javascript
"use strict";

module.exports = {
  services: {
    cipher: {},
    hash: {},
    image: {},
    location: {},
    mailer: {},
    payment: {},
    pusher: {},
    sms: {},
    social: {},
    storage: {}
  }
}
```

**Comment by ghaiklor** [comment]
@mikedevita you can achieve the same behavior with default Sails setup. Just create `config/local.js` file and it overrides any properties you declare there. There is no needs for implementing such kind of features.

http://sailsjs.com/documentation/concepts/configuration/the-local-js-file

**Comment by mikedevita** [comment]
Does it override or does it merge?

**Comment by ghaiklor** [comment]
@mikedevita it overrides and merges. If property exists, it will be overridden, otherwise it will be merged.

**Comment by mikedevita** [comment]
@ghaiklor that idea wont work as is because you call `config/services/*.js` independently and don't hook into `sails.config.services.*` so either there needs to be a change in the config to use `sails.config.services` or use merge like i suggested.

**Comment by ghaiklor** [comment]
@mikedevita yeah, I see, makes sense. It will be great to get rid of direct requiring of configuration files and use `sails.config`.

**Comment by mikedevita** [comment]
Ill look into this and submit another PR..

**Comment by mikedevita** [comment]
a bit of an update, by making the services functions you can then obtain access to `sails.config`

e.g;

**api/services/CipherService.js**
```javascript
module.exports = {
  jwt: (config) => cipher('jwt', _.merge({}, sails.config.services.cipher.jwt, config))
}
```

doing this you can then change anywhere `CipherService.jwt.encodeSync()` to be `CipherService.jwt().encodeSync()` I am not sure how to modify the yo generators to include this new syntax though. So any help would be appreciated there.
